pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1791,3 +1801,80 @@ func (c *Cluster) releaseOrchestrationLock() {
 	c.orchestrationMutex.Unlock()
 	logger.Debugf("Released lock for mon orchestration")
 }
+
+func (c *Cluster) RotateMonCephxKeys() (bool, error) {
+	shouldRotateMonKeys, err := c.shouldRotateMonKeys()
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to check if cephx keys for mon daemons in the namespace %q should be rotated", c.ClusterInfo.Namespace)
+	}
+
+	if !shouldRotateMonKeys {
+		logger.Debugf("cephx key rotation for mon daemon in the namespace %q is not required", c.ClusterInfo.Namespace)
+		return shouldRotateMonKeys, nil
+	}
+
+	logger.Infof("cephx keys for mon daemons in the namespace %q will be rotated", c.ClusterInfo.Namespace)
+
+	k := keyring.GetSecretStore(c.context, c.ClusterInfo, c.ClusterInfo.OwnerInfo)
+	newKey, err := k.RotateKey(controller.MonCephxUser)
+	if err != nil {
+		return shouldRotateMonKeys, errors.Wrapf(err, "failed to rotate cephx key for mon daemon in the namespace %q", c.ClusterInfo.Namespace)
+	}
+
+	c.ClusterInfo.MonitorSecret = newKey
+
+	// update the mon-secret key in the cluster access secret
+	err = controller.UpdateClusterAccessSecret(c.context.Clientset, c.ClusterInfo, c.ownerInfo)
+	if err != nil {
+		return shouldRotateMonKeys, errors.Wrapf(err, "failed to update the rook-ceph-mon secret after rotating the mon cephx keys in the namespace %q", c.ClusterInfo.Namespace)
+	}
+
+	// update the keyring which all mons share
+	if _, err := k.CreateOrUpdate(keyringStoreName, c.genMonSharedKeyring()); err != nil {
+		return shouldRotateMonKeys, errors.Wrapf(err, "failed to save mon keyring secret after rotating the mon cephx keys in the namespace %q", c.ClusterInfo.Namespace)
+	}
+
+	logger.Infof("successfully rotated cephx keys for mon daemons in the namespace %q", c.ClusterInfo.Namespace)
+
+	return shouldRotateMonKeys, nil
+}
+
+func (c *Cluster) shouldRotateMonKeys() (bool, error) {
+	clusterObj := &cephv1.CephCluster{}
+	if err := c.context.Client.Get(c.ClusterInfo.Context, c.ClusterInfo.NamespacedName(), clusterObj); err != nil {
+		return false, errors.Wrapf(err, "failed to get cluster %v.", c.ClusterInfo.NamespacedName())
+	}
```

Comment:
No need to Get() this here when we are already doing a CephCluster reconcile.

Reply:
Not sure if I followed that. We still need to `Get()` the cephcluster object to compare the `cephxConfig` with `cephxStatus` and determine if the key should be rotated or not.

Reply:
I finally get what you're saying here. We have the spec, but the status isn't readily available. While we could pass it down, it would be through so many methods that it's not really worth it. Ignore this/these comments